README.md

```diff
@@ -87,6 +87,47 @@ The Ginkgo CUDA module has the following __additional__ requirements:
 The Ginkgo OMP module has the following __additional__ requirements:
 *  _MinGW_ or _Cygwin_
 
+Depending on the configuration settings, some manual work might be required:
+* Build Ginkgo as shared library:
+  Add `PROJECT_BINARY_DIR/GINKGO_WINDOWS_SHARED_LIBRARY_RELPATH` into the environment variable `PATH`.
+  `GINKGO_WINDOWS_SHARED_LIBRARY_RELPATH` is `windows_shared_library` by default. More Details are available in the [Installation page](./INSTALL.md).
+  * cmd: `set PATH="<PROJECT_BINARY_DIR/GINKGO_WINDOWS_SHARED_LIBRARY_RELPATH>;%PATH%"`
+  * powershell: `$env:PATH="<PROJECT_BINARY_DIR/GINKGO_WINDOWS_SHARED_LIBRARY_RELPATH>;$env:PATH"`
+
+  CMake will gives the error message including `<PROJECT_BINARY_DIR/GINKGO_WINDOWS_SHARED_LIBRARY_RELPATH>` if the path is not correct.
+  ```Did not find this build in the environment variable PATH. Please add <path> into the environment variable PATH.```
+* Build Ginkgo with Debug mode:
+  1. `bigobj` issue: encontering `too many sections` needs `\bigobj` or `-Wa,-mbig-obj`
+  2. `ld` issue: encontering `ld: error: export ordinal too large` needs `-O1`
+
+  The following are the details for different environments:
+  * _Microsoft Visual Studio_:
+    1. `bigobj` issue
+      * `cmake -DCMAKE_CXX_FLAGS=\bigobj <other parameters> <source_folder>` which might overwrite the default settings.
+      * add `\bigobj` into the environment variable `CXXFLAGS` (only availible in the first cmake configuration)
+        * cmd: `set CXXFLAGS=\bigobj`
+        * powershell: `$env:CXXFLAGS=\bigobj`
+    2. `ld` issue (_Microsoft Visual Studio_ does not have this issue)
+  * _Cygwin_:
+    1. `bigobj` issue
+      * add `-Wa,-mbig-obj -O1` into the environment variable `CXXFLAGS` (only availible in the first cmake configuration)
+        * `export CXXFLAGS="-Wa,-mbig-obj -O1"`
+      * `cmake -DCMAKE_CXX_FLAGS=-Wa,-mbig-obj <other parameters> <source_folder>`, which might overwrite the default settings.
+    2. `ld` issue (If build Ginkgo as static library, do not need this)
+      * `cmake -DGINKGO_COMPILER_FLAGS="-Wpedantic -O1" <other parameters> <source_folder>` (`GINKGO_COMPILER_FLAGS` is `-Wpedantic` by default)
+  * _MinGW_:
+    1. `bigobj` issue
+      * add `-Wa,-mbig-obj -O1` into the environment variable `CXXFLAGS` (only availible in the first cmake configuration)
+        * cmd: `set CXXFLAGS="-Wa,-mbig-obj"`
+        * powershell: `$env:CXXFLAGS="-Wa,-mbig-obj"`
+      * `cmake -DCMAKE_CXX_FLAGS=-Wa,-mbig-obj <other parameters> <source_folder>`, which might overwrite the default settings.
+    2. `ld` issue (If build Ginkgo as static library, do not need this)
+      * `cmake -DGINKGO_COMPILER_FLAGS="-Wpedantic -O1" <other parameters> <source_folder>` (`GINKGO_COMPILER_FLAGS` is `-Wpedantic` by default)
```

Comment:
Since this is fairly lengthy and detailed, whereas `README.md` is supposed to be a summary of everything, wouldn't it be better to put this into `INSTALL.md` and link the section here, with a link such as `[windows build detailed instructions](./INSTALL.md#section)`

Potentially, more of the windows specific issues could be put there, but as a minimum these 20 lines or so are good candidates.